src/main/java/org/broadinstitute/dsm/jobs/TestBostonUPSTrackingJob.java

```diff
@@ -58,36 +59,71 @@ public void execute(JobExecutionContext context) throws JobExecutionException {
+                                        rs.getString(DBConstants.DSM_KIT_REQUEST_ID),
+                                        rs.getString("kit." + DBConstants.KIT_LABEL),
+                                        rs.getString("kit." + DBConstants.DSM_TRACKING_TO),
+                                        rs.getString("kit." + DBConstants.TRACKING_RETURN_ID),
+                                        rs.getString("kit." + DBConstants.ERROR),
+                                        rs.getString("kit." + DBConstants.MESSAGE),
+                                        rs.getString("kit." + DBConstants.DSM_RECEIVE_DATE),
+                                        rs.getString("kit." + DBConstants.UPS_TRACKING_STATUS),
+                                        rs.getString("kit." + DBConstants.UPS_TRACKING_DATE),
+                                        rs.getString("kit." + DBConstants.UPS_RETURN_STATUS),
+                                        rs.getString("kit." + DBConstants.UPS_RETURN_DATE),
+                                        rs.getString("req." + DBConstants.COLLABORATOR_PARTICIPANT_ID),
+                                        rs.getString("req." + DBConstants.EXTERNAL_ORDER_NUMBER),
+                                        rs.getBoolean("kit." + DBConstants.CE_ORDER)
+                                );
+                                String type;
+                                if (StringUtils.isNotBlank(kit.getTrackingToId())) {
+                                    type = kit.getUpsTrackingStatus();
+                                    if (StringUtils.isNotBlank(type)) {// get only type from it
+                                        type = type.substring(0, type.indexOf(' '));
+                                    }
+                                    if (!"D".equals(type)) {//don't include delivered ones
+                                        try {
+                                            updateKitStatus(conn, kit, false);
+                                        } catch (Exception e) {
+                                            logger.error("Could not update outbound status for " + kit.getExternalOrderNumber(), e);
+                                        }
+                                    }
                                 }
-                            }
-                        }
-                        kits = ids.get("return");
-                        if (kits != null) {
-                            logger.info("checking return status for " + kits.size() + " tracking numbers");
-                            for (DdpKit kit : kits.values()) {
-                                try {
-                                    updateKitStatus(kit, true);
-                                } catch (Exception e) {
-                                    logger.error("Could not update return status for " + kit.getExternalOrderNumber(), e);
+                                if (StringUtils.isNotBlank(kit.getTrackingReturnId())) {
```

Comment:
Similar comment here--by putting the defensive parse/null checks into `DdpKit.isDelivered()` (or expanding to two methods like `hasArrived()` and `hasReturned()`, you can centralize this, test it, and not have to duplicate it in other places.